### PR TITLE
Use Redis.new rather than Redis.current

### PR DESCRIPTION
**Pull request by edwardkerry**
Redis.current will be deprecated in Redis 5.0.

As Redis.current ultimately calls Redis.new we can migrate this method now to pave the way for future upgrades.

From Redis changelog from [5.0][1]:

‘Removed Redis.current. You shouldn't assume there is a single global Redis connection, use a connection pool instead, and libaries using Redis should accept a Redis instance (or connection pool) as a config. E.g. MyLibrary.redis = Redis.new(…).’

[Redis.current][2] just calls Redis.new so should just be able to swap methods.

[trello](https://trello.com/c/oGFWhrCO/1551-migrate-away-from-deprecated-redis-method)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: config/brakeman.ignore [hunk 1]
```diff
@@ -29,6 +29,9 @@
       },
       "user_input": "params[:list].presence",
       "confidence": "High",
+      "cwe_id": [
```
Comment (1pretz1) [inline config/brakeman.ignore:32]: Are these brakeman changes related?

Comment (edwardkerry) [reply config/brakeman.ignore:32]: These changes are not directly related but an artefact of re-running Brakeman to update the fingerprint of our ignoring the use of `Redis.current|new`. I'm happy to remove them if they are confusing?

File: config/brakeman.ignore [hunk 2]
```diff
@@ -76,18 +83,21 @@
...
       "warning_code": 0,
-      "fingerprint": "d3d42c84d0b0acf57ccb1c9d1133ca43389fd725d0471016013e52bf59269b83",
+      "fingerprint": "814bc0a997725602aa2913dd902c8d77c28de9044d4562f60a898ad250d04b5a",
       "check_name": "SQL",
       "message": "Possible SQL injection",
       "file": "lib/csv_report_generator.rb",
       "line": 11,
       "link": "https://brakemanscanner.org/docs/warning_types/sql_injection/",
-      "code": "Redis.current.lock(\"publisher:#{Rails.env}:report_generation_lock\", :life => 15.minutes)",
+      "code": "Redis.new.lock(\"publisher:#{Rails.env}:report_generation_lock\", :life => 15.minutes)",
```
Comment (kevindew) [inline config/brakeman.ignore:100]: Shall we change the code so these no longer produce warnings. 

I can't see there being any need for us to have Rails.env inside these locking keys - it's hard to think of a scenario where you'd need that distinction. So we could just simplify them to get rid of needing to update this further.

Comment (edwardkerry) [reply config/brakeman.ignore:100]: Would you object to my making this a new card on our backlog? There are some similar/identical brakeman warnings in other apps, and it's not really connected to the intention of this PR, so might make sense to complete the Redis work and then tackle the Brakeman work en masse after.

File: config/brakeman.ignore (hunk 1 above)
Comment (1pretz1) [reply config/brakeman.ignore:32]: Sorry for the delayed response! I'm happy for the changes to be part of this PR, I just wondered why they are present. Thanks for explaining!

File: config/brakeman.ignore (hunk 2 above)
Comment (KludgeKML) [reply config/brakeman.ignore:100]: @kevindew just trying to clear up some of Ed's cards while he's off - are you okay with his suggestion (make this change a new one in our backlog separate from the redis fix)?

Comment (KludgeKML) [reply config/brakeman.ignore:100]: I see Kev's away, I'm going to say we can expedite this and catch the other work Ed mentions in this card: https://trello.com/c/84yWQ0NJ/1662-clean-up-brakeman-errors-following-on-from-rediscurrent-redisnew-updates

Comment (ollietreend) [reply config/brakeman.ignore:100]: Yep – that sounds good to me @KludgeKML 👍🏻
